File: backend/Api.Data/DeploymentStatus.cs
using System.Text.Json.Serialization;

namespace Api.Data;

[JsonConverter(typeof(JsonStringEnumConverter))]
public enum DeploymentStatus
{
    Unknown = 0,
    Deploying = 1,
    Completed = 2,
    Queued = 3,
    Failed = 4,
    Cancelled = 5,
}

File: frontend/src/lib/deployment-status.ts
export enum DeploymentStatus {
    Unknown = "Unknown",
    Deploying = "Deploying",
    Completed = "Completed",
    Queued = "Queued",
    Failed = "Failed"
}

export function getDeploymentStatusFromNumber(stateNum?: number): DeploymentStatus {
    switch (stateNum) {
        case 0: return DeploymentStatus.Unknown;
        case 1: return DeploymentStatus.Deploying;
        case 2: return DeploymentStatus.Completed;
        case 3: return DeploymentStatus.Queued;
        case 4: return DeploymentStatus.Failed;
        default: return DeploymentStatus.Unknown;
    }
}

